pdk/1.1/modules/ROOT/pages/policies-pdk-configure-features-timer.adoc

```diff
@@ -0,0 +1,161 @@
+= Configuring Delayed, Periodic, and Synchronous Functions
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+PDK provides the `Clock` injectable to build a timer to enable delayed, periodic, and synchronous functions. Each policy only supports one timer.
+
+Each timer provides the following two functions:
+
+[source,rust]
+----
+pub async fn next_tick(&self) -> bool;
+pub async fn sleep(&self, interval: Duration) -> bool;
+----
+
+* `next_tick`: Sleeps for the minimum amount of time the timer is capable of distinguishing.
+* `sleep`: Sleeps for an amount of ticks where the elapsed time is greater or equal than the provided `Duration`. For example, if you configure the period of the clock to 10 seconds, calling `timer.sleep(Duration::from_secs(1))` sleeps for 10 seconds.
+
+Both functions return `true` when the specified time has passed or false if the policy is unapplied or edited during execution.
+
+== Configure the Timer
+
+To inject a `timer` in your policy, insert a `Clock` into your policy configuration function and define a timer with a time period:
+
+[source,rust]
+----
+#[entrypoint]
+async fn configure(
+   launcher: Launcher,
+   Configuration(bytes): Configuration,
+   clock: Clock, // Inject the clock to be able to launch async tasks.
+   client: HttpClient,
+) -> Result<()> {
+
+
+   // set the period between ticks of the timer.
+   let timer = clock.period(Duration::from_secs(10));
+----
+
+After you inject the timer, share a reference of the `timer` with the `on_request` and `on_response` functions to begin creating delayed functions:
+
+[source,rust]
+----
+    launcher
+        .launch(on_request(|rs| request_filter(rs, &timer, &config)))
+        .await?;
+----
+
+You can now await timer functions in `on_request` and `on_response` functions, for example:
```

Comment:
Should timer be in mono font?

Reply:
OK, it's just that it's in mono font on line 24.

Reply:
Ahh, yeah, that's confusing. I'll change the above.